Update template api repository for Sandbox/Incubated approach

hdamker opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* enhancement/feature

#### What this PR does / why we need it:

Adding badge and text alternatives for 
* Independent Sandbox API Repositories
* Sandbox API Repositories within a Sub Project
* Incubated API Repositories (always within a Sub Project)
* Graduated API Repositories (same as Incubated, other badge and and one word replaced)
* Working Groups (the template has to be further adapted to the specific Working Group scope)

See https://github.com/camaraproject/Governance/blob/main/ProjectStructureAndRoles.md for further context

#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes #13 

#### Special notes for reviewers:

hdamker committed:
Update pull_request_template.md

hdamker committed:
Update README.MD

hdamker committed:
Update README.md for Sandbox/Incubated API Repositories

hdamker commented:
@jgarciahospital @MarkusKuemmerle would be great if you can have a short view on the changes

MarkusKuemmerle commented:
@hdamker : The option for graduated repos is missing in the "repo name" section. Add it?

hdamker committed:
Update README.md - added Graduation

A Graduated API Repository was previously an Incubated one ... therefore only badge and one word need to be replaced within this change.

hdamker commented:
> @hdamker : The option for graduated repos is missing in the "repo name" section. Add it?

We will never create a repository directly as "Graduated". But I have added line 43 in https://github.com/camaraproject/Template_API_Repository/pull/14/commits/d8bcda82af8842e0a39ab269884b8527bb09eca4 to explain the two little change to "graduate" an Incubated API Repository.

BTW: we can think about adding a line with "Incubated: <Month> <Year>" after the line "Started: ..." ... but that's maybe more a topic for the Lifecycle documentation.
